tests/integration/test_pytest_extension.py

```diff
@@ -25,6 +25,19 @@ def test_example(snapshot):
         """
     testdir.makepyfile(conftest=conftest)
     testdir.makepyfile(test_file=testcase)
+    return testdir
+
+
+@pytest.mark.skipif(pytest_version < (6, 0), reason="at least pytest 6 required")
+def test_ignores_non_function_nodes(run_testcases):
+    testdir = run_testcases
     result = testdir.runpytest("test_file.py", "-v", "--snapshot-update")
     result.stdout.re_match_lines((r".*test_file.py::CUSTOM.*"))
     assert result.ret == 0
+
+
+def test_handles_pyargs_non_module_when_both_given(run_testcases):
+    result = run_testcases.runpytest(
+        "-v", "test_file.py", "--pyargs", "test_file", "--snapshot-update"
+    )
+    assert result.ret == 0
```

Comment:
Would fail with `ModuleNotFoundError: No module named 'test_file.py'; 'test_file' is not a package` if the `Exception` was not handled.